src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceUpdateActor.scala

```diff
@@ -4,29 +4,35 @@ package core.task.tracker.impl
 import java.time.Clock
 import java.util.concurrent.TimeoutException
 
-import akka.actor.{Actor, Props, Status}
+import akka.Done
+import akka.actor.{Actor, ActorRef, Props, Status}
 import akka.event.LoggingReceive
 import akka.pattern.pipe
+import akka.util.Timeout
 import com.typesafe.scalalogging.StrictLogging
 import mesosphere.marathon.core.instance.Instance
-import mesosphere.marathon.core.task.tracker.impl.InstanceUpdateActor.{ActorMetrics, FinishedInstanceOp, ProcessInstanceOp}
+import mesosphere.marathon.core.instance.update.{InstanceUpdateEffect, InstanceUpdateOpResolver}
+import mesosphere.marathon.core.task.tracker.impl.InstanceTrackerActor.UpdateContext
+import mesosphere.marathon.core.task.tracker.impl.InstanceUpdateActor.{ActorMetrics, FinishedUpdate, QueuedUpdate}
 import mesosphere.marathon.metrics._
 
 import scala.collection.immutable.Queue
-import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
 
 object InstanceUpdateActor {
-  def props(clock: Clock, metrics: ActorMetrics, processor: InstanceOpProcessor): Props = {
-    Props(new InstanceUpdateActor(clock, metrics, processor))
+  def props(clock: Clock, metrics: ActorMetrics,
+    instanceTrackerRef: ActorRef,
+    stateOpResolver: InstanceUpdateOpResolver,
```

Comment:
This makes so much more sense to me now. I actually inject such logic with a trait to make it testable: https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/core/task/jobs/impl/ExpungeOverdueLostTasksActor.scala#L20. What do you think?